spec/index.bs

```diff
@@ -4426,40 +4435,84 @@ typedef double GPUPipelineConstantValue; // May represent WGSL's bool, f32, i32,
-                        :: the |binding| is a non-comparison sampler
+                        : {{GPUSamplerBindingType/"filtering"}} or {{GPUSamplerBindingType/"non-filtering"}}
+                        :: |variable| has type `sampler`.
                         : {{GPUSamplerBindingType/"comparison"}}
-                        :: the |binding| is a comparison sampler
+                        :: |variable| has type `comparison_sampler`.
                     </dl>
 
                 : {{GPUBindGroupLayoutEntry/texture}}
-                :: If |entry|.{{GPUBindGroupLayoutEntry/texture}}.{{GPUTextureBindingLayout/multisampled}} is:
+                ::
+                    If, and only if,
+                    |entry|.{{GPUBindGroupLayoutEntry/texture}}.{{GPUTextureBindingLayout/multisampled}}
+                    is `true`, |variable| has type `texture_multisampled_2d<T>` or `texture_depth_multisampled_2d<T>`.
+                ::
+                    If |entry|.{{GPUBindGroupLayoutEntry/texture}}.{{GPUTextureBindingLayout/sampleType}} is:
                     <dl class=switch>
-                        : `true`
-                        :: the |binding| is a multisampled texture.
-                        : `false`
-                        :: The |binding| is a sampled texture with a sample count of 1.
+                        : {{GPUTextureSampleType/"float"}}, {{GPUTextureSampleType/"unfilterable-float"}},
+                            {{GPUTextureSampleType/"sint"}} or {{GPUTextureSampleType/"uint"}}
+                        ::
+                            |variable| has type `texture_1d<T>`, `texture_2d<T>`, `texture_2d_array<T>`,
+                            `texture_cube<T>`, `texture_cube_array<T>`, `texture_3d<T>`, or
+                            `texture_multisampled_2d<T>`.
+                        ::
+                            If |entry|.{{GPUBindGroupLayoutEntry/texture}}.{{GPUTextureBindingLayout/sampleType}} is:
+                            <dl class=switch>
+                                : {{GPUTextureSampleType/"float"}} or {{GPUTextureSampleType/"unfilterable-float"}}
+                                :: The sampled type `T` is `f32`.
+                                : {{GPUTextureSampleType/"sint"}}
+                                :: The sampled type `T` is `i32`.
+                                : {{GPUTextureSampleType/"uint"}}
+                                :: The sampled type `T` is `u32`.
+                            </dl>
+
+                        : {{GPUTextureSampleType/"depth"}}
+                        ::
+                            |variable| has type `texture_depth_2d`, `texture_depth_2d_array`,
+                            `texture_depth_cube`, `texture_depth_cube_array`, or `texture_depth_multisampled_2d`.
+                    </dl>
+                ::
+                    If |entry|.{{GPUBindGroupLayoutEntry/texture}}.{{GPUTextureBindingLayout/viewDimension}} is:
```

Comment:
The way it's described now is in 3 rules that have intersecting possibilities.
Would it be significantly easier to define this from the opposite direction entirely? I.e. if the shader type is `texture_1d<T>` then `multisampled == false` and `dimension == "1d"`. If `T` is `f32`, then the sample type is "float" or "unfilterable-float".

The benefit here is - if I'm reading through the rules trying to understand why something doesn't work, I'd know exactly where to look, i.e. the aspects to check are separated better.

Reply:
I mostly had it this way because it matched the original style, but yeah, I think that would be good

Reply:
I think to do this we'd need to flip the entire switch starting at the root, so it's a big change. I also think that doing so would still result in rules with intersecting possibilities.

Anyway, it's a big change, so I'll leave it for later.